shared/utility/math/filter/UKF.hpp

```diff
@@ -71,11 +75,10 @@ namespace utility::math::filter {
          * @brief Generate new sigma points given our mean and covariance.
          */
         template <typename T, int S>
-        static Eigen::Matrix<T, S, NUM_SIGMA_POINTS> generate_sigma_points(const Eigen::Matrix<T, S, 1>& mean,
-                                                                           const Eigen::Matrix<T, S, S>& covariance,
-                                                                           const T& sigma_weight) {
-
-            Eigen::Matrix<T, S, NUM_SIGMA_POINTS> points;
+        static Eigen::ComputationInfo generate_sigma_points(const Eigen::Matrix<T, S, 1>& mean,
```

Comment:
Every function that calls this function acts on the return value from this function, so I feel like this is a moot point

Reply:
It would be for future users of the function. If you shouldn't call this function without using the return value, then it should be `[[nodiscard]]`. There's no real downside

Reply:
@Bidski please make this `[[nodiscard]]`